### PR TITLE
fix: space in data-group record names

### PR DESCRIPTION
**Pull request by joelkeener**
When adding a record to a data-group, it will create multiple records from a name with spaces, and it will only add data to the last record. 

This will create a single record, preserving the original name with its spaces.

### CODE REVIEW

File: src/storageDataGroup.js [hunk 1]
```diff
@@ -161,7 +161,7 @@ function readDataGroup(path) {
 function outputToObject(output) {
     const jsonString = output
         .replace(/ltm data-group internal .*? {/, '{')
-        .replace(/(\s*)(.*?) {/g, '$1"$2" : {')
+        .replace(/(\s*)("?)(.*?)("?) {/g, '$1"$3" : {')
```
Comment (ansmithf5) [inline src/storageDataGroup.js:164]: Why was this changed to $3?

**Review by tpitkinf5 — CHANGES_REQUESTED** [review]
We should probably add a unit test help verify that the fix works and that we don't introduce a regression in the future.

File: src/storageDataGroup.js (hunk 1 above)
Comment (ansmithf5) [reply src/storageDataGroup.js:164]: It's because the $3 is referencing the ().

Comment (joelkeener) [reply src/storageDataGroup.js:164]: Yup, and $2 is now the match of zero or one double-quotes that should be before the name.

**Comment by joelkeener** [comment]
> We should probably add a unit test help verify that the fix works and that we don't introduce a regression in the future.

This is done now.

**Review by tpitkinf5 — APPROVED** [review]
Looks great! Thanks for fixing this corner case!

**Review by ansmithf5 — APPROVED** [review]
Wow the GitHub workflow is really wonky when you're used to GitLab. Yeah this looks good Joel :)
